packages/expo-doctor/src/doctor.ts

```diff
@@ -0,0 +1,103 @@
+import { getConfig } from '@expo/config';
+import chalk from 'chalk';
+
+import { ExpoConfigSchemaCheck } from './checks/ExpoConfigSchemaCheck';
+import { GlobalPrereqsVersionCheck } from './checks/GlobalPrereqsVersionCheck';
+import { IllegalPackageCheck } from './checks/IllegalPackageCheck';
+import { InstalledDependencyVersionCheck } from './checks/InstalledDependencyVersionCheck';
+import { SupportPackageVersionCheck } from './checks/SupportPackageVersionCheck';
+import { DoctorCheck, DoctorCheckParams } from './checks/checks.types';
+import { gteSdkVersion } from './utils/gteSdkVersion';
+import { logNewSection } from './utils/ora';
+import { warnUponCmdExe } from './warnings/windows';
+
+export async function runCheckAsync(
+  check: DoctorCheck,
+  checkParams: DoctorCheckParams
+): Promise<boolean> {
+  try {
+    const ora = logNewSection(check.description);
+    const result = await check.runAsync(checkParams);
+    if (!result.isSuccessful) {
+      ora.fail();
+      if (result.issues.length) {
+        console.log(chalk.underline(chalk.yellow(`Issues:`)));
+        console.group();
+        for (const issue of result.issues) {
+          console.log(chalk.yellow(`${issue}`));
+        }
+        console.groupEnd();
+      }
+      if (result.advice.length) {
+        console.log(chalk.underline(chalk.green(`Advice:`)));
+        console.group();
+        for (const advice of result.advice) {
+          console.log(chalk.green(`• ${advice}`));
+        }
+        console.groupEnd();
+      }
+      return false;
+    } else {
+      ora.succeed();
+      return true;
+    }
+  } catch (e: any) {
+    if (e.code === 'ENOTFOUND') {
+      console.log(
+        chalk.red(
+          `Error: this check requires a connection to the Expo API. Please check your network connection.`
+        )
+      );
+    } else {
+      console.log(chalk.red(`Error: check could not be completed due to an unexpected error.`));
+    }
+    return false;
+  }
+}
+
+export async function actionAsync(projectRoot: string) {
+  await warnUponCmdExe();
+
+  const { exp, pkg } = getConfig(projectRoot);
+
+  // expo-doctor relies on versioned CLI, which is only available for 44+
+  if (!gteSdkVersion(exp, '44.0.0')) {
+    console.log(
+      chalk.red(
+        `expo-doctor supports Expo SDK +44. Use 'expo-cli doctor' for SDK 43 and lower.`
+      )
+    );
+    process.exitCode = 1;
+    return;
+  }
+
+  // add additional checks here
+  const checks = [
+    new GlobalPrereqsVersionCheck(),
+    new IllegalPackageCheck(),
+    new SupportPackageVersionCheck(),
+    new InstalledDependencyVersionCheck(),
+    new ExpoConfigSchemaCheck(),
+  ];
+
+  let foundSomeIssues = false;
+
+  const checkParams = { exp, pkg, projectRoot };
+
+  for (const check of checks) {
+    if (!(await runCheckAsync(check, checkParams))) {
+      foundSomeIssues = true;
+    }
```

Comment:
Similar to my comment about `warnAbout...` it would be great if running the checks, and outputting the result were separated. In my opinion, `runCheckAsync` should just return the results ({ isSuccessful, issues, advice}), and you can properly parse the results here (isSuccessful) and output issues and advice.

This would also work a lot better if you want to parallellize these checks in the future. Logging during checking would mix up the output of different checks together, while running them in parallel and having a single threaded loop to output the results doesn't have this issue.

To me, this is a serious architecture concern throughout.